include/boost/test/impl/execution_monitor.ipp

```diff
@@ -1298,7 +1307,7 @@ execution_monitor::execute( boost::function<int ()> const& F )
 #if defined(BOOST_NO_TYPEID) || defined(BOOST_NO_RTTI)
                               "unknown boost::exception" ); }
 #else
-                              boost::diagnostic_information(ex).c_str() ); }
+                              "%s", boost::diagnostic_information(ex).c_str() ); }
```

Comment:
I am not sure to understand why this change would be needed.

Reply:
It's a security issue as well as a correctness issue:

gcc.compile.c++ ../../../bin.v2/libs/test/build/gcc-7/debug/link-static/threading-multi/visibility-hidden/execution_monitor.o
In file included from ../../../libs/test/src/execution_monitor.cpp:16:0:
../../../boost/test/impl/execution_monitor.ipp: In member function ‘int boost::execution_monitor::execute(const boost::function<int()>&)’:
../../../boost/test/impl/execution_monitor.ipp:1311:73: error: format not a string literal and no format arguments [-Werror=format-security]
                               boost::diagnostic_information(ex).c_str() ); }
                                                                         ^
cc1plus: some warnings being treated as errors

I'll add a test illustrating this.